Zend/zend_ast.c

```diff
@@ -951,6 +964,7 @@ static void* ZEND_FASTCALL zend_ast_tree_copy(zend_ast *ast, void *buf)
 		zend_ast *new = (zend_ast*)buf;
 		new->kind = ast->kind;
 		new->attr = ast->attr;
+		new->lineno = ast->lineno;
```

Comment:
I'm not sure if `lineno` is intentionally not copied here. I couldn't see any hacks that repurpose `lineno`. If this is fine maybe we should do the same for the other ast types above.

Reply:
According to Nikita it's unintentional